packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DashboardGridLayoutWithAddCardPanel.tsx

```diff
@@ -47,6 +47,40 @@ export class DashboardGridLayoutWithAddCardPanel extends BaseComponent<
     };
   }
 
+  public componentDidMount(): void {
+    if (this._cardsForAddCardPanel !== this.props.addCardPanelCards || this._cardsForLayout !== this.props.dashboardCards) {
+      this._cardsForAddCardPanel = this.props.addCardPanelCards;
+      this._cardsForLayout = this.props.dashboardCards;
+      const cardIds: string[] = [];
+      const cardNodes: JSX.Element[] = [];
```

Comment:
Can you briefly write as to what the issues were and what the fixes for those are. I am not able to understand what you are exactly trying to solve.